apis/r/R/QueryCondition.R

```diff
@@ -0,0 +1,316 @@
+#  MIT License
+#
+#  Copyright (c) 2021-2024 TileDB Inc.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to deal
+#  in the Software without restriction, including without limitation the rights
+#  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#  copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in all
+#  copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#  SOFTWARE.
+
+# ================================================================
+#' Create a 'tiledbsoma_query_condition' object from an expression
+#'
+#' The grammar for query conditions is constrained to the operators
+#' (\code{">"}, \code{">="}, \code{"<"}, \code{"<="}, \code{"=="},
+#' \code{"!="}, \code{"%in%"}, \code{"%nin%"}), and three boolean operators
+#' (\code{"&&"}, also as \code{"&"}, (\code{"||"}, also as \code{"|"}, and
+#' \code{"!"} for negation.  Note that we locally define \code{"%nin%"} as
+#' \code{Negate()} call around \code{%in%)} which extends R a little for this
+#' use case.
+#'
+#' Expressions, in the R language syntax, are parsed locally by this function.
+#'
+#' @param expr An expression that is understood by the TileDB grammar for
+#' query conditions.
+#'
+#' @param schema The Arrow schema for the array for which a query
+#' condition is being prepared. This is necessary to obtain type information
+#' about left-hand sides of query expressions.
+#'
+#' @param strict A boolean toogle to, if set, errors if a non-existing
+#' attribute is selected or filtered on, defaults to 'TRUE'; if 'FALSE' a
+#' warning is shown but execution proceeds.
+#'
+#' @param somactx SOMAContext pointer.
+#'
+#' @return A `tiledbsoma_query_condition` object.
+#' @export
+parse_query_condition_new <- function(
```

Comment:
Keep this internal to make it easier to re-build/re-write at a later date
```suggestion
#' @return A `tiledbsoma_query_condition` object.
#' 
#' @noRd
#'
parse_query_condition_new <- function(
```